modules/standard/Random.chpl

```diff
@@ -1154,41 +1154,91 @@ module Random {
         return _choice(this, x, size=size, replace=replace, prob=prob);
       }
 
-      /* Randomly shuffle a 1-D array. */
-      proc shuffle(arr: [?D] ?eltType ) {
-
-        if D.rank != 1 then
-          compilerError("Shuffle requires 1-D array");
-
-        const low = D.alignedLow,
-              stride = abs(D.stride);
+      pragma "no doc"
+      inline proc _fisherYatesShuffle(arr : [] eltType, start : int, blockSize : int) {
```

Comment:
@rahulghangas do we mind making "no doc" pragma `proc`s private and remove the `pragma`? But since other functions in this module uses the "no doc" pragma I think it would be reasonable to use it here as well. What do you think?

Reply:
I would prefer "no doc", just in case someone might want to use a fast serial shuffle (although I don't see why).

Reply:
@ben-albrecht thoughts?

Reply:
I agree that we should make this private, but if we think a fast serial shuffle would be valuable, we should explore creating a public UI for that. I would suggest opening a separate issue for collecting feedback on that and pursuing it after this PR is merged to keep it manageable.

As a simple straw proposal, I could see that getting exposed through another argument to shuffle, e.g. `shuffle(A, parallel=false)`

Reply:
Re making it private - we don't support private methods yet so we have to use "no doc" right now as a workaround.

Reply:
Whoops - I didn't catch that this was a method, and not a function. Thanks @mppf.